src/mds/MDLog.h

```diff
@@ -301,5 +307,10 @@ class MDLog {
   std::set<LogSegment*> expired_segments;
   std::set<LogSegment*> expiring_segments;
   uint64_t events_since_last_major_segment = 0;
+
+  // -- events --
+  ceph::coarse_mono_time start_time;
+  uint64_t start_pos, end_pos;
```

Comment:
These can be fetched from the journaler. No need to save here.

Reply:
@batrick don't we need the start position? It will change as the replay progresses right?

Reply:
> @batrick don't we need the start position?

The start pos is `mdlog->get_journaler()->get_trimmed_pos()`.

> Actual read position will change as the replay progresses right?

Yes.

>  I also assumed that actual write position will change since there will be appends as well. Let me know if I am wrong.

Not during replay.